src/Model/ISO2ROI.py

```diff
@@ -0,0 +1,292 @@
+import pydicom, platform, os.path, datetime
+from pathlib import Path
+from pydicom.dataset import FileDataset, FileMetaDataset
+from skimage import measure
+from PySide6 import QtCore, QtWidgets, QtGui
+from PySide6.QtWidgets import QMessageBox
+from src.Controller.PathHandler import resource_path
+from src.Model import ImageLoading
+from src.Model import ROI
+from src.Model.Isodose import get_dose_grid
+from src.Model.PatientDictContainer import PatientDictContainer
+from src.Model.GetPatientInfo import DicomTree
+
+
+class WorkerSignals(QtCore.QObject):
+    signal_roi_drawn = QtCore.Signal(tuple)
+    signal_save_confirmation = QtCore.Signal()
+
+class ISO2ROI:
+    """This class is for converting isodose levels to ROIs."""
+    def __init__(self):
+        self.worker_signals = WorkerSignals()
+        self.signal_roi_drawn = self.worker_signals.signal_roi_drawn
+        self.signal_save_confirmation = \
+            self.worker_signals.signal_save_confirmation
+        self.signal_save_confirmation.connect(self.save_confirmation)
+        self.advised_save_rtss = False
+        self.isodose_levels = {}
+
+    def start_conversion(self, interrupt_flag, progress_callback):
+        """
+        Goes the the steps of the iso2roi conversion.
+        :param interrupt_flag: interrupt flag to stop process
+        :param progress_callback: signal that receives the current
+                                  progress of the loading.
+        """
+        patient_dict_container = PatientDictContainer()
+
+        progress_callback.emit(("Validating Datasets", 0))
+        val = ImageLoading.is_dataset_dicom_rt(patient_dict_container.dataset)
+
+        # Stop loading
+        if interrupt_flag.is_set():
+            print("Stopped ISO2ROI")
+            return False
+
+        if not val:
```

Comment:
what does `val` refer to?

Reply:
`val` is a boolean; if it is True, it means that a loaded dataset/patient does not have at least one each of RT Dose, RT Plan, RT Struct, and CT Image. I understand that it is named poorly however and this will be fixed.

Reply:
Change has been made and added to the PR.